configmap-reload.go

```diff
@@ -67,6 +76,109 @@ func init() {
 	prometheus.MustRegister(requestsByStatusCode)
 }
 
+// generateNewConf returns a fully rendered configuration file with support
+// for golang templating
+func generateNewConf(templatePath string) (string, error) {
+	rel, err := filepath.Rel(volumeDirs[0], templatePath)
+	if err != nil {
+		return "", err
+	}
+
+	// if it's not the template file we want, then we just return the raw contents
+	// this also doubles as not running without a template file being set.
+	if rel != *templateFile {
+		b, err := ioutil.ReadFile(templatePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to read file: %v", err)
+		}
+		return string(b), nil
+	}
+
+	log.Printf("rendering file '%s'", templatePath)
+
+	t := template.New("conf")
+
+	b, err := ioutil.ReadFile(templatePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file: %v", err)
+	}
+
+	tmpl, err := t.Parse(string(b))
+	if err != nil {
+		return "", fmt.Errorf("failed to create template: %v", err)
+	}
+
+	var inf interface{}
```

Comment:
This could very-well be over engineering, but I'll throw out this thought anyway: What if we created an interface which was responsible for getting the templating context/data/whatever you wanna call it, passed that in as an argument to this function, and had main choose which instantiation (env or file) to wire up with it?

Reply:
It would certainly be much cleaner, I'll see if I can get around to doing that before merging this